File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;

/**
 * 品牌服务层接口
 */
public interface BrandService {

    /**
     * 返回全部品牌列表
     *
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 返回分页列表
     *
     * @param pageNum  当前页码
     * @param pageSize 当前页数量
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);

    /**
     * 添加品牌
     *
     * @param tbBrand
     */
    public void add(TbBrand tbBrand);

    /**
     * 修改品牌
     *
     * @param tbBrand
     */
    public void update(TbBrand tbBrand);

    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    public TbBrand findOne(Long id);

    /**
     * 根据数组ids批量删除品牌
     *
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 带条件分页
     *
     * @return
     */
    public PageResult findPage(TbBrand tbBrand, int pageNum, int pageSize);
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService', function ($http) {

    // 全查品牌数据
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    };

    // 分页品牌
    this.findPage = function (page, rows) {
        return $http.get('../brand/findPage.do?page=' + page + '&rows=' + rows);
    };

    // 添加品牌
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    };

    // 更新品牌
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    };

    // 单查品牌
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    };

    // 删除品牌
    this.delete = function (ids) {
        return $http.get('../brand/delete.do?id=' + ids);
    };

    // 带条件分页查询品牌
    this.search = function (page, rows, searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&rows=' + rows, searchEntity);
    };
});